subsys/gazell/Kconfig

```diff
@@ -34,6 +34,28 @@ config GAZELL_PAIRING_HOST
 	bool "Host"
 	depends on ENTROPY_GENERATOR
 
+choice GAZELL_PAIRING_CONFIG
```

Comment:
Maybe we could avoid the choice and simply unconditionally define the included file path as:

```
config GAZELL_PAIRING_CONFIG_FILE_NAME
	string "Gazell Pairing configuration file name"
	default "gzp_config.h"
        help
          Modify the file name to use an application-specific Gazell Pairing configuration file.
```

Reply:
I wonder if the approach selected here is proper.
If we keep a configuration file with selected name under a few various include paths used during our build, we may end up with accidentally using a wrong file (which is not expected).
Still, I would not block if the solution is acceptable for Gazell codeowners.

Reply:
The GAZELL_PAIRING_CONFIG_FILE_NAME option is more concise. Good suggestion.

Some options in the GZP header file are byte arrays, which is difficult to convert to Kconfig options. Frankly, I am not clear how some options can be fine tuned. It is why I leave it as it is. :sweat:

Reply:
I also do not have a suggestion how can we switch to Kconfigs here (and since it's not HW configuration, using DTS here may not be clear approach too). Maybe we can stick to updating header file name then.

Reply:
Remove choice `GAZELL_PAIRING_CONFIG`.
Leave `GAZELL_PAIRING_CONFIG_USER` and define `GAZELL_PAIRING_CONFIG_USER_FILE`.

Reply:
> GAZELL_PAIRING_CONFIG_FILE_NAME

Good suggestion for config name and description. I'd prefer leave user config file name different than default one. So that user will not be confused which header is actual used.

Reply:
In this case, we may align with mbedTLS configuration. Suggest naming the Kconfig options:
- GAZELL_PAIRING_USER_CONFIG_ENABLE
- GAZELL_PAIRING_USER_CONFIG_FILE